master/docs/manual/worker-transition.rst

```diff
@@ -0,0 +1,1091 @@
+Complete removal of "slave"-containing terminology is planned in version
+**TODO**.
+
+Rename of API introduced in beta versions of Buildbot 0.9.0 done without
+providing fallback.
+
+Old names fallback settings
+---------------------------
+
+Use of obsolete names will raise Python warnings with category
+:py:exc:`buildbot.worker_transition.DeprecatedWorkerAPIWarning`.
+By default these warnings are printed in the application log.
+This behaviour can be changed by setting appropriate Python warnings settings
+via Python's :py:mod:`warnings` module:
+
+.. code-block:: python
+
+    import warnings
+    from buildbot.worker_transition import DeprecatedWorkerAPIWarning
+    # Treat old-name usage as errors:
+    warnings.simplefilter("error", DeprecatedWorkerAPIWarning)
+
+See Python's :py:mod:`warnings` module documentation for complete list of
+available actions, in particular warnings can be disabled using
+``"ignore"`` action.
+
+It's recommended to configure warnings inside :file:`buildbot.tac`, before
+using any other Buildbot classes.
+
+Changed API
+-----------
+
+In general "Slave" and "Buildslave" parts in identifiers and messages were
+replaced with "Worker"; "SlaveBuilder" with "WorkerForBuilder".
+
+.. todo::
+
+    * This list will be updated along with actual changing of the API.
+
+    * Most of this list can be generated/verified by grepping use of
+      ``worker_transition`` helpers.
+
+    * Some of attribute/methods that were renamed may be actually private.
+      If they are private, then no fallback should be provided and they
+      change shouldn't be documented.
+
+    * Thorough tests for old modules imports are not yet written
+      (import of old-named modules is done in tests, but expected warning for
+      deprecated module import is not tested yet).
+
+    * Test that module reloading works and doesn't produce more warnings than
+      it should.
+
+    * Some classes are marked as ``(private?)`` because they are not mentioned
+      in the documentation, but in my opinion may be used by
+      end users (so they either should be documented, or fallback for them
+      should be removed).
+      If some identifiers don't need fallback for old name, then it should be
+      removed.
+
+Here is the list of changed API (use of old names from this list will work):
+
+.. list-table::
+   :header-rows: 1
+
+   * - Old name
+     - New name
+
+   * - :py:class:`buildbot.interfaces.IBuildSlave` (private?)
+     - :py:class:`~buildbot.interfaces.IWorker`
+
+
+   * - :py:class:`buildbot.interfaces.NoSlaveError` (private?)
+     - left as is, but deprecated (it shouldn't be used at all)
+
+
+   * - :py:class:`buildbot.interfaces.BuildSlaveTooOldError`
+     - :py:class:`~buildbot.interfaces.WorkerTooOldError`
+
+
+   * - :py:class:`buildbot.interfaces.LatentBuildSlaveFailedToSubstantiate`
+       (private?)
+     - :py:class:`~buildbot.interfaces.LatentWorkerFailedToSubstantiate`
```

Comment:
For most of these, I suspect they are private, but there's a chance that someone is using them, so keeping the fallback makes sense.  Maybe in the final version of this document, the introduction to this list could include, "Note that some of these symbols are not included in Buildbot's public API.  Compatibility is provided as a convenience to those using the private symbols anyway."